[UnifiedPDF] PDFPageNumberIndicator cannot use UIVisualEffectView in all configurations

aprotyas opened the pull request:
#### 9a293216b4744c279c10e6c93c52307c10bce484
<pre>
[UnifiedPDF] PDFPageNumberIndicator cannot use UIVisualEffectView in all configurations
<a href="https://bugs.webkit.org/show_bug.cgi?id=292454">https://bugs.webkit.org/show_bug.cgi?id=292454</a>
<a href="https://rdar.apple.com/149981713">rdar://149981713</a>

Reviewed by Aditya Keerthi and Richard Robinson.

UIVisualEffectView is not appropriate for the backdrop of the page
number indicator under certain platform configurations. In such cases,
this patch introduces a path to fallback to a generic UIView that has
its background appropriately configured to not regress the indicator&apos;s
functionality.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Modules/Internal/WebKitInternal.h:
* Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.h:
* Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm:
(-[WKPDFPageNumberIndicator initWithFrame:view:pageCount:]):

Note that when the backdrop is not a UIVisualEffectView, we must take
care to add the page number label directly to the backdrop view, and not
to the `contentView` under the backdrop.

(-[WKPDFPageNumberIndicator backdropView]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/294462@main">https://commits.webkit.org/294462@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b59fa95be1ec307ffde349a2a0a1e0988c7bedc5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101887 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21555 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11871 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/107046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/52521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21863 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30062 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/107046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/52521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104894 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/16900 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92002 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/107046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/16727 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/51876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/86574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10100 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109409 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29020 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/21372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/86551 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29381 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88206 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86128 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30885 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8608 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/23191 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16574 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28948 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34243 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28759 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32082 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30318 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.h:
```diff
@@ -35,10 +35,19 @@
 
 @property (nonatomic) unsigned currentPageNumber;
 @property (nonatomic) unsigned pageCount;
+@property (nonatomic, readonly, strong) UIView *backdropView;
```
Nit: `strong` is unnecessary.

aprotyas replied:
Fixed

pxlcoder commented on Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm:
```diff
@@ -71,7 +72,11 @@ - (instancetype)initWithFrame:(CGRect)frame view:(WKWebView *)view pageCount:(si
+        _backdropView = adoptNS([[UIVisualEffectView alloc] initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]]);
+    else
+        _backdropView = adoptNS([[UIView alloc] init]);
+    [self configureBackgroundForBackdropViewIfNeeded];
```
This can be inside the `else`?

aprotyas replied:
Indeed, moved!

aprotyas committed:
[UnifiedPDF] PDFPageNumberIndicator cannot use UIVisualEffectView in all configurations
https://bugs.webkit.org/show_bug.cgi?id=292454
rdar://149981713

Reviewed by Aditya Keerthi and Richard Robinson.

UIVisualEffectView is not appropriate for the backdrop of the page
number indicator under certain platform configurations. In such cases,
this patch introduces a path to fallback to a generic UIView that has
its background appropriately configured to not regress the indicator's
functionality.

* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Modules/Internal/WebKitInternal.h:
* Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.h:
* Source/WebKit/UIProcess/PDF/WKPDFPageNumberIndicator.mm:
(-[WKPDFPageNumberIndicator initWithFrame:view:pageCount:]):

Note that when the backdrop is not a UIVisualEffectView, we must take
care to add the page number label directly to the backdrop view, and not
to the `contentView` under the backdrop.

(-[WKPDFPageNumberIndicator backdropView]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/294462@main
